### PR TITLE
Remove VArray for async RGL

### PR DESCRIPTION
**Pull request by prybicki**

### CODE REVIEW

File: src/graph/SpatialMergePointsNode.cpp [hunk 1]
```diff
@@ -67,13 +67,9 @@ void SpatialMergePointsNode::enqueueExecImpl()
...
-		data->resize(width, false, false);
-		size_t offset = 0;
 		for (const auto& input : pointInputs) {
-			size_t pointCount = input->getPointCount();
 			const auto toMergeData = input->getFieldData(field);
-			data->insertData(toMergeData->getReadPtr(MemLoc::Device), pointCount, offset);
-			offset += pointCount;
+			data->appendFrom(toMergeData);
 		}
 	}
```
Comment (msz-rai) [inline src/graph/SpatialMergePointsNode.cpp:74]: We should clear `data` before the second for-loop to delete data from the previous graph run.
As of now, it also merges points temporally.

File: src/graph/TransformPointsNode.cpp [hunk 2]
```diff
@@ -18,23 +18,22 @@
...
+	const auto* inputPtr = inputField->getReadPtr();
+	auto* outputPtr = output->getWritePtr();
 	gpuTransformPoints(getStreamHandle(), pointCount, inputPtr, outputPtr, transform);
 }
 
-VArray::ConstPtr TransformPointsNode::getFieldData(rgl_field_t field)
+IAnyArray::ConstPtr TransformPointsNode::getFieldData(rgl_field_t field)
 {
 	if (field == XYZ_F32) {
 		CHECK_CUDA(cudaStreamSynchronize(nullptr));
```
Comment (msz-rai) [inline src/graph/TransformPointsNode.cpp:31]: To be removed I guess

File: src/memory/ArrayImpl.inl [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+template<typename T>
+void Array<T>::copyFromExternal(const T *src, size_t srcCount) {
+	this->resize(srcCount, false, false);
+
+	auto dstStreamBound = std::dynamic_pointer_cast<IStreamBound>(shared_from_this());
+	CudaStream::Ptr copyStream = dstStreamBound != nullptr
+	                             ? dstStreamBound->getStream()
+	                             : CudaStream::getNullStream();
+	CHECK_CUDA(cudaMemcpyAsync(this->getRawWritePtr(), src, srcCount * sizeof(T), cudaMemcpyDefault, copyStream->getHandle()));
```
Comment (msz-rai) [inline src/memory/ArrayImpl.inl:23]: If `this` is host you can use `memcpy` to make it faster.

File: test/src/graphCaseTest.cpp [hunk 4]
```diff
@@ -324,21 +324,21 @@ TEST_F(GraphCase, FormatNodeResults)
...
+// *
+// * This test uses CUDA API to get current total memory usage of the GPU,
+// * which makes this test inevitably prone to random fluctuations.
+// *
+// * In the future this might fixed by wrapping all CUDA allocations and exposing API to internally track memory usage.
+// *
+// * In RGL API we could give user control over which memory is used by this node.
+// * However, this raises question, why not allow it for all nodes, which
+// * summons another jungle of complexity, which I'd prefer to avoid until it is really necessary.
+// */
```
Comment (msz-rai) [inline test/src/graphCaseTest.cpp:341]: Comment leftover

**Review by msz-rai — COMMENTED** [review]
GJ :1st_place_medal: 
A few comments with suggestions / possible bugs.

**Comment by prybicki** [comment]
Fixed everything.

**Review by msz-rai — APPROVED** [review]

**Pull request merged**